src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -528,11 +533,16 @@ uct_ib_mlx5_devx_reg_mt(uct_ib_mlx5_md_t *md, void *address, size_t length,
                         uint64_t access_flags, uint32_t *mkey_p,
                         uct_ib_mlx5_devx_ksm_data_t **ksm_data_p)
 {
-    size_t chunk = md->super.config.mt_reg_chunk;
-    int mr_num   = ucs_div_round_up(length, chunk);
+    size_t chunk_size    = md->super.config.mt_reg_chunk;
+    size_t atomic_prefix = (uintptr_t)address % md->super.dev.atomic_align;
+    void* prefix_address = (void*)((uintptr_t)address - atomic_prefix);
+    size_t align_padding = ucs_padding((uintptr_t)prefix_address, chunk_size);
     uct_ib_mlx5_devx_ksm_data_t *ksm_data;
     ucs_status_t status;
     int dmabuf_fd;
+    int mr_num;
+
+    length += atomic_prefix;
```

Comment:
seems it didn't exist in previous code, is that a bug fix?

Reply:
Before this patch MT registration worked successfully only for buffers aligned to `mt_reg_chunk` so `atomic_prefix` was always 0.